froide/helper/templatetags/breadcrumb_helper.py

```diff
@@ -1,31 +1,10 @@
 from django import template
-from django.urls import NoReverseMatch, reverse
-
-register = template.Library()
 
+from froide.helper.breadcrumbs import Breadcrumbs
 
-def normalize_breadcrumb(breadcrumb):
-    if type(breadcrumb) == tuple:
-        if type(breadcrumb[1]) == str:
-            try:
-                breadcrumb = (breadcrumb[0], reverse(breadcrumb[1]))
-            except NoReverseMatch:
-                pass
-
-        return breadcrumb
-    else:
-        return (breadcrumb, None)
+register = template.Library()
 
 
 @register.simple_tag(takes_context=True)
 def get_breadcrumbs(context, view=None):
-    if hasattr(view, "get_breadcrumbs") and callable(view.get_breadcrumbs):
-        return view.get_breadcrumbs(context)
-
-    if hasattr(view, "breadcrumbs"):
-        return map(normalize_breadcrumb, view.breadcrumbs)
-
-
-@register.filter
-def has_link(value):
-    return type(value) == tuple and len(value) == 2
+    return Breadcrumbs.from_view(view, context)
```

Comment:
`from_view` expects a non-null view, should we remove support vor `get_breadcrumbs` with a `None`-view?

Reply:
Or add

```python
if view is None:
    return None
```

Reply:
Not sure what you mean. `from_view` is able to handle `view = None`, and returns `None` in that case